src/OtpInput/useOtpInput.ts

```diff
@@ -24,6 +25,9 @@ export const useOtpInput = ({
   };
 
   const handleTextChange = (value: string) => {
+    if (type === "alpha" && /[^a-zA-Z]/.test(value)) return;
+    if (type === "numeric" && /[^\d]/.test(value)) return;
+    if (type === "alphanumeric" && /[^a-zA-Z\d]/.test(value)) return;
```

Comment:
What about this way of writing the same logic? To make it cleaner and more scalable
`
const regexMap = {
    alpha: /[^a-zA-Z]/,
    numeric: /[^\d]/,
    alphanumeric: /[^a-zA-Z\d]/
};

if (regexMap[type] && regexMap[type].test(value)) return;
`

Reply:
@anday013 Nice suggestion, done! 🚀